rust/core-lib/src/tabs.rs

```diff
@@ -559,16 +561,35 @@ impl CoreState {
 
             if autodetect_whitespace {
                 let indentation = Indentation::parse(editor.borrow().get_buffer());
+                let line_ending = LineEnding::parse(editor.borrow().get_buffer());
+                
                 if let Ok(Some(indentation)) = indentation {
                     let changes = match indentation {
                         Indentation::Tabs => json!({"translate_tabs_to_spaces": false}).as_object().unwrap().to_owned(),
                         Indentation::Spaces(n) => json!({"translate_tabs_to_spaces": true, "tab_size": n}).as_object().unwrap().to_owned()
                     };
                     
-                    // Paste values from new config into pre-existing
+                    // Paste values from new config into pre-existing if config update succeeds
                     let new_conf = self.config_manager.table_for_update(ConfigDomain::UserOverride(buffer_id), changes);
-                    for (k, v) in new_conf.iter() {
-                        config.insert(k.to_owned(), v.to_owned());
+                    if self.config_manager.set_user_config(ConfigDomain::UserOverride(buffer_id), new_conf.clone()).is_ok() {
+                        for (k, v) in new_conf.iter() {
+                            config.insert(k.to_owned(), v.to_owned());
+                        }
+                    }
+                }
+
+                if let Ok(Some(line_ending)) = line_ending {
+                    let changes = match line_ending {
+                        LineEnding::Windows => json!({"line_ending": "\r\n"}).as_object().unwrap().to_owned(),
+                        LineEnding::Linux => json!({"line_ending": "\n"}).as_object().unwrap().to_owned(),
+                        LineEnding::Mac => json!({"line_ending": "\r"}).as_object().unwrap().to_owned()
+                    };
+
+                    let conf = self.config_manager.table_for_update(ConfigDomain::UserOverride(buffer_id), changes);
+                    if self.config_manager.set_user_config(ConfigDomain::UserOverride(buffer_id), conf.clone()).is_ok() {
+                        for (k, v) in conf.iter() {
+                            config.insert(k.to_owned(), v.to_owned());
+                        }
```

Comment:
Would it be possible to combine these config updates into a single loop? I'm not very happy at iterating over the entire config for every changed setting...

Reply:
We've discussed this on IRC, but I want to put it here just to clarify:
The variable naming is not clear enough here, as this only iterates through the diff generated by the `changes` variable, which in this code is 2 settings max.